### PR TITLE
Add textlint checker

### PR DESCRIPTION
**Pull request by kisaragi-hiu**
This integrates a modified version of @robstewart57's [A textlint flycheck checker in Emacs](https://www.macs.hw.ac.uk/~rs46/posts/2018-12-29-textlint-flycheck.html) into flycheck. I first packaged it in https://github.com/kisaragi-hiu/flycheck-textlint; that repo will be turned into a tutorial if this gets merged.

As textlint supports multiple formats, I have to add a new option to let users add their own mode → textlint plugin associations, otherwise textlint would just fail with a backtrace.

This implements #899.

### CODE REVIEW

File: flycheck.el [hunk 1]
```diff
@@ -10516,6 +10517,69 @@ See URL `http://www.gnu.org/software/texinfo/'."
...
+                    #'booleanp
+                    (mapcar
+                     (lambda (item)
+                       (let ((mode (car item))
+                             (plugin (cdr item)))
+                         (if (or (and mode (booleanp mode)) ; mode is t
+                                 (derived-mode-p mode))
+                             plugin
+                           nil)))
+                     flycheck-textlint-plugin-alist))))
```
Comment (fmdkdd) [inline flycheck.el]: What about
```emacs-lisp
(cdr
 (-first
   (pcase-lambda (`(,mode . ,plugin))
     (or (and (booleanp mode) mode) ; mode is t
         (derived-mode-p mode)))
   flycheck-textlint-plugin-alist))
```
instead?

**Comment by kai2nenobu** [comment]
This is a very attractive feature :smile:. I hope this feature will be merged ASAP.

BTW, I have one comment about a lint result parsing. 

`textlint` can output a lint result as JSON and its format is compatible with a `eslint` format.  Fortunately, `flycheck` already has the parser function of `eslint` result (`flycheck-parse-eslint`). If `flycheck-parse-eslint` is used with `:error-parser`, `flycheck` may handle error results more precisely (severity, rule id, multi-line message), I think.

Checker implementation is something like [this](https://github.com/kai2nenobu/.emacs.d/blob/ba1bc1a240045e1bcda0ce86bf932f4155b4409c/my-lisp/flycheck-textlint.el). (My own emacs configuration repository)

File: flycheck.el (hunk 1 above)
Comment (kisaragi-hiu) [reply flycheck.el]: I didn't know about `-first`, thanks!

I get a warning that `pcase-lambda` depends on Emacs 25, so I've implemented it with a plain lambda / let.

File: test/flycheck-test.el [hunk 2]
```diff
@@ -4584,6 +4584,13 @@ The manifest path is relative to
    '(9 nil warning "printindex before document beginning: @printindex cp"
        :checker texinfo)))
 
+(flycheck-ert-def-checker-test textlint (text markdown) nil
+  (let ((flycheck-textlint-config "language/text/textlintrc.json"))
+    (flycheck-ert-should-syntax-check
+      "language/text/text.txt" '(text-mode markdown-mode)
```
Comment (fmdkdd) [inline test/flycheck-test.el]: Please re-indent these two lines to pass the CI.

File: flycheck.el (hunk 1 above)
Comment (fmdkdd) [reply flycheck.el]: Oh, good call.

**Comment by fmdkdd** [comment]
LGTM.  Please squash your commits and I well merge.

**Comment by fmdkdd** [comment]
Thank you for this nice work, and for iterating quickly on this PR.  :+1:

**Pull request merged**